src/narrativelog/routers/find_messages.py

```diff
@@ -101,89 +102,127 @@ async def find_messages(
         default=None,
         description="Primary software components names or fragments of names. "
         "All messages with a component that matches any of these are included. "
         "Repeat the parameter for each value. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_path' instead.",
     ),
     exclude_primary_software_components: None
     | list[str] = fastapi.Query(
         default=None,
         description="Primary software components names or fragments of names. "
         "All messages with a component that matches any of these are excluded. "
         "Repeat the parameter for each value. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_path' instead.",
     ),
     primary_hardware_components: None
     | list[str] = fastapi.Query(
         default=None,
         description="Primary hardware components names or fragments of names. "
         "All messages with a component that matches any of these are included. "
         "Repeat the parameter for each value. "
-        "Each entry should be a valid component name entry on the OBS jira project.",
+        "Each entry should be a valid component name entry on the OBS jira project. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_path' instead.",
     ),
     exclude_primary_hardware_components: None
     | list[str] = fastapi.Query(
         default=None,
         description="Primary hardware components names or fragments of names. "
         "All messages with a component that matches any of these are excluded. "
-        "Repeat the parameter for each value.",
+        "Repeat the parameter for each value. "
+        "**This field is deprecated and will be removed in v1.0.0**. "
+        "Please use 'components_path' instead.",
+    ),
+    components_path: None
+    | str = fastapi.Query(
+        default=None,
+        description="Components structure in JSON format."
+        "All messages with a 'components_json' field that "
+        "matches the specified structure are included."
+        "The structure represents the hierarchy of "
+        "systems and subsystems on the OBS jira project. "
+        "Current structure: `{'systems': ['system1', ..., 'systemN'], "
+        "'subsystems': ['subsystem1', ..., 'subsystemN'], "
+        "'components': ['component1', ..., 'componentN']}`. "
+        "E.g. `{'systems': ['AuxTel'], 'subsystems': ['ATCamera']}` will match"
+        " all messages that have 'AuxTel' as system and 'ATCamera' as subsystem.",
```

Comment:
Ah, THIS is was I was expecting to see earlier.
But its not clear to me what the parameter value would look like for "component_path" search.  Will the pieces be combined with AND? With OR?  (I don't know where the implementation is for that kind of stuff with fastAPI.  Hoping it uses ORM to combine instead of SQL.)

Reply:
Good catch! I was expecting the example to implicitly indicate how the search will work, i.e. `all messages that have 'AuxTel' as system and 'ATCamera' as subsystem.` meaning that we are combining key-values over AND. I'll iterate over this to provide something more similar to what we have before with "include any of" and "exclude any of" :ok_hand:.